src/QsCompiler/Compiler/RewriteSteps/ClassicallyControlled.cs

```diff
@@ -44,6 +44,8 @@ public bool PreconditionVerification(QsCompilation compilation)
                 .ToHashSet();
             var requiredBuiltIns = new HashSet<QsQualifiedName>()
             {
+                BuiltIn.NoOp.FullName,
```

Comment:
Do we also need to add `IndexRange` to this list, or is that already handled as a dependency somewhere else?

Reply:
The Classically Controlled logic doesn't rely on `IndexRange`, so it doesn't need to be listed here.